src/shogun/features/DenseFeatures.h

```diff
@@ -134,6 +137,27 @@ template<class ST> class CDenseFeatures: public CDotFeatures
 	 */
 	SGVector<ST> get_feature_vector(int32_t num) const;
 
+	/**
+	 * Returns an iterator to the first vector of features.
+	 */
+	iterator begin() noexcept
```

Comment:
This code is the same for all feature classes right?
Maybe we can move it somewhere (mixin?)

Reply:
done!